Test crashes after chooseHost() fails due to live checksum lookup

```
go generate ./pkg/... ./cmd/...
go fmt ./pkg/... ./cmd/...
go vet ./pkg/... ./cmd/...
go run vendor/sigs.k8s.io/controller-tools/cmd/controller-gen/main.go all
CRD manifests generated under '/home/rbryant/go/src/github.com/metalkube/cluster-api-provider-baremetal/config/crds' 
RBAC manifests generated under '/home/rbryant/go/src/github.com/metalkube/cluster-api-provider-baremetal/config/rbac' 
kustomize build config/default/ > provider-components.yaml
2019/03/22 14:20:31 Adding nameprefix and namesuffix to Namespace resource will be deprecated in next release.
echo "---" >> provider-components.yaml
cd vendor && kustomize build sigs.k8s.io/cluster-api/config/default/ >> ../provider-components.yaml
2019/03/22 14:20:31 Adding nameprefix and namesuffix to Namespace resource will be deprecated in next release.
go test ./pkg/... ./cmd/... -coverprofile cover.out
?   	github.com/metalkube/cluster-api-provider-baremetal/pkg/apis	[no test files]
?   	github.com/metalkube/cluster-api-provider-baremetal/pkg/apis/baremetal	[no test files]
ok  	github.com/metalkube/cluster-api-provider-baremetal/pkg/apis/baremetal/v1alpha1	10.400s	coverage: 23.7% of statements
2019/03/22 14:20:34 looking for checksum for http://172.22.0.1/images/redhat-coreos-maipo-latest.qcow2 at http://172.22.0.1/images/redhat-coreos-maipo-latest.qcow2.md5sum
--- FAIL: TestChooseHost (30.01s)
    actuator_test.go:121: Get http://172.22.0.1/images/redhat-coreos-maipo-latest.qcow2.md5sum: dial tcp 172.22.0.1:80: i/o timeout
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x158 pc=0xe7a24a]

goroutine 20 [running]:
testing.tRunner.func1(0xc000236700)
	/usr/lib/golang/src/testing/testing.go:792 +0x387
panic(0xf4fe20, 0x1aea420)
	/usr/lib/golang/src/runtime/panic.go:513 +0x1b9
github.com/metalkube/cluster-api-provider-baremetal/pkg/cloud/baremetal/actuators/machine.TestChooseHost(0xc000236700)
	/home/rbryant/go/src/github.com/metalkube/cluster-api-provider-baremetal/pkg/cloud/baremetal/actuators/machine/actuator_test.go:123 +0x7ea
testing.tRunner(0xc000236700, 0x10d37e8)
	/usr/lib/golang/src/testing/testing.go:827 +0xbf
created by testing.(*T).Run
	/usr/lib/golang/src/testing/testing.go:878 +0x35c
FAIL	github.com/metalkube/cluster-api-provider-baremetal/pkg/cloud/baremetal/actuators/machine	30.030s
?   	github.com/metalkube/cluster-api-provider-baremetal/pkg/controller	[no test files]
?   	github.com/metalkube/cluster-api-provider-baremetal/pkg/webhook	[no test files]
?   	github.com/metalkube/cluster-api-provider-baremetal/cmd/manager	[no test files]
make: *** [Makefile:17: unit] Error 1
```

Note this line that causes the failure:
```
    actuator_test.go:121: Get http://172.22.0.1/images/redhat-coreos-maipo-latest.qcow2.md5sum: dial tcp 172.22.0.1:80: i/o timeout
```

--- a/pkg/cloud/baremetal/actuators/machine/actuator.go
+++ b/pkg/cloud/baremetal/actuators/machine/actuator.go
@@ -19,11 +19,8 @@ package machine
 import (
 	"context"
 	"fmt"
-	"io/ioutil"
 	"log"
 	"math/rand"
-	"net/http"
-	"strings"
 	"time"
 
 	bmh "github.com/metalkube/baremetal-operator/pkg/apis/metalkube/v1alpha1"
@@ -235,27 +232,13 @@ func (a *Actuator) chooseHost(ctx context.Context, machine *machinev1.Machine) (
 		Namespace: machine.Namespace,
 	}
 
-	// FIXME(dhellmann): The Stein version of Ironic supports passing
-	// a URL. When we upgrade, we can stop doing this work ourself.
-	log.Printf("looking for checksum for %s at %s",
-		instanceImageSource, instanceImageChecksumURL)
-	resp, err := http.Get(instanceImageChecksumURL)
-	if err != nil {
-		return nil, err
-	}
-	defer resp.Body.Close()
-	instanceImageChecksum, err := ioutil.ReadAll(resp.Body)
-	if err != nil {
-		return nil, err
-	}
-
 	// FIXME(dhellmann): When we stop using the consts for these
 	// settings, we need to pass the right values.
 	chosenHost.Spec.Image = &bmh.Image{
 		URL:      instanceImageSource,
-		Checksum: strings.TrimSpace(string(instanceImageChecksum)),
+		Checksum: instanceImageChecksumURL,
 	}
-	err = a.client.Update(ctx, chosenHost)
+	err := a.client.Update(ctx, chosenHost)
 	if err != nil {
 		return nil, err
 	}

